File: extension/src/main/java/com/thornbirds/component/view/IComponentView.java
package com.thornbirds.component.view;

/**
 * This class defines basic interface of Component View in MVP Pattern for android platform.
 *
 * @author YangLi yanglijd@gmail.com
 * @see IEventView
 */
public interface IComponentView<PRESENTER, VIEW extends IViewProxy>
        extends IEventView<PRESENTER, VIEW> {
}

File: extension/src/main/java/com/thornbirds/component/view/IComponentView.kt
package com.thornbirds.component.view

import android.view.View

/**
 * Basic interface of View Proxy
 *
 * @author YangLi yanglijd@gmail.com
 */
interface IViewProxy {
    /**
     * Returns the real view of the ViewProxy.
     *
     * @param <T> type of real view
     * @return real view
     */
    fun <T : View> getRealView(): T
}

/**
 * This class defines basic interface of Component View in MVP Pattern for android platform.
 *
 * @see IEventView
 * @author YangLi yanglijd@gmail.com
 */
interface IComponentView<PRESENTER, VIEW : IViewProxy> : IEventView<PRESENTER, VIEW>